gradle-plugin/src/main/kotlin/io/reflekt/plugin/util/Util.kt

```diff
@@ -0,0 +1,20 @@
+package io.reflekt.plugin.util
+
+import org.gradle.api.Project
+import org.gradle.api.file.SourceDirectorySet
+import org.gradle.api.internal.HasConvention
+import org.gradle.api.tasks.SourceSet
+import org.gradle.api.tasks.SourceSetContainer
+import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
+
+internal inline fun <reified T : Any> Project.myExtByName(name: String): T = extensions.getByName(name) as T
+
+internal val Project.mySourceSets: SourceSetContainer
+    get() = myExtByName("sourceSets")
+
+val SourceSet.kotlin: SourceDirectorySet
+    get() =
+        (this as HasConvention)
+            .convention
```

Comment:
We should change it in the future (in another PR), we used this function since we had a hot deadline and had to make it as soon as possible and I knew only this way. But if you want to fix it in this PR I can try. But this PR is very big and probably wi should make it later

Reply:
okay, let's leave a TODO at least?